File: Java&Kotlin lecture/src/main/java/com/java/student/Student.java
package com.java.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    static int PASS = 60;

    public Student(String id, String name, int english, int math) {
        this.id = id;
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void showScore() {
        int average = getAverage();
        System.out.println(name + "\tEnglish:" + english + "\tMath:" + math);
        System.out.print(average > PASS ? "PASS" : "FAILED");

        char grading = 'F';

//        if(average >=90 && average <= 100){
//            grading = 'A';
//        } else if (average >=80 && average <= 89) {
//            grading = 'B';
//        } else if (average >=70 && average <= 79) {
//            grading = 'C';
//        } else if (average >=60 && average <= 69) {
//            grading = 'D';
//        } else {
//            grading = 'E';
//        }

        switch (average / 10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'E';
                break;
        }

        System.out.println("\t" + grading);
    }

    public void checkName() {
        System.out.println(name.length());
    }

    public int getHighest() {
        if(english >= math) {
            System.out.print("English ");
            return english;
        }else {
            System.out.print("Math ");
            return math;
        }
    }

    public int getAverage() {
        return (english + math) / 2;
    }
}

File: Java&Kotlin lecture/src/main/kotlin/com/kotlin/student/Student.kt
package com.kotlin.student

import java.util.*

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)

    println("Please enter student's name:")
    val name = scanner.next()

    println("Please enter student's English score:")
    val english = scanner.nextInt()

    println("Please enter student's Math score:")
    val math = scanner.nextInt()

    val student = Student("01", name, english, math)
    student.showScore()
    student.checkName()
    println("High score: ${student.getHighest()}")
    println("Average score: ${student.getAverage()}")

    val gStudent = GraduatedStudent("100", "Jane", 85, 67, 88);
    gStudent.showScore()
    gStudent.checkName()
    println("High score: ${gStudent.getHighest()}")
    println("Average score: ${gStudent.getAverage()}")
}

open class Student(val id: String, var name: String? , var english: Int, var math: Int) {
    companion object {
        @JvmStatic  // 存取時不用再透過companion才能呼叫PASS
        var PASS = 60
    }

    open fun showScore() {
        val average = getAverage()
        println("$name\tEnglish:$english\tMath:$math")
        println(if(average > PASS) "PASS" else "FAILED")

        val grading: Char = when (average) {
            in 90 .. 100 -> 'A'
            in 80 .. 89 -> 'B'
            in 70 .. 79 -> 'C'
            in 60 .. 69 -> 'D'
            else -> 'E'
        }

        println("\t" + grading)
    }

    fun checkName() {
        println(name?.length)
    }

    open fun getHighest(): Int {
        return if(english > math){
            print("English ")
            english
        }else {
            print("Math ")
            math
        }
    }

    open fun getAverage(): Int = (english + math) / 2
}